File: app/src/main/java/com/zeyad/rxredux/screens/BaseFragment.java
package com.zeyad.rxredux.screens;

import com.zeyad.rxredux.core.redux.BaseViewModel;
import com.zeyad.rxredux.snackbar.SnackBarFactory;

import android.view.View;
import android.widget.Toast;

/**
 * @author by ZIaDo on 7/21/17.
 */

public abstract class BaseFragment<S, VM extends BaseViewModel<S>>
        extends com.zeyad.rxredux.core.redux.BaseFragment<S, VM> {

    public void showToastMessage(String message) {
        showToastMessage(message, Toast.LENGTH_LONG);
    }

    public void showToastMessage(String message, int duration) {
        Toast.makeText(getContext(), message, duration).show();
    }

    /**
     * Shows a {@link android.support.design.widget.Snackbar} message.
     *
     * @param message An string representing a message to be shown.
     */
    public void showSnackBarMessage(View view, String message, int duration) {
        if (view != null) {
            SnackBarFactory.getSnackBar(SnackBarFactory.TYPE_INFO, view, message, duration).show();
        } else {
            throw new IllegalArgumentException("View is null");
        }
    }

    public void showSnackBarWithAction(String typeSnackBar, View view, String message, String actionText,
            View.OnClickListener onClickListener) {
        if (view != null) {
            SnackBarFactory.getSnackBarWithAction(typeSnackBar, view, message, actionText, onClickListener).show();
        } else {
            throw new IllegalArgumentException("View is null");
        }
    }

    public void showSnackBarWithAction(String typeSnackBar, View view, String message, int actionText,
            View.OnClickListener onClickListener) {
        showSnackBarWithAction(typeSnackBar, view, message, getString(actionText), onClickListener);
    }

    /**
     * Shows a {@link android.support.design.widget.Snackbar} errorResult message.
     *
     * @param message  An string representing a message to be shown.
     * @param duration Visibility duration.
     */
    public void showErrorSnackBar(String message, View view, int duration) {
        if (view != null) {
            SnackBarFactory.getSnackBar(SnackBarFactory.TYPE_ERROR, view, message, duration).show();
        } else {
            throw new IllegalArgumentException("View is null");
        }
    }
}

File: app/src/main/java/com/zeyad/rxredux/screens/BaseFragment.kt
package com.zeyad.rxredux.screens

import android.os.Parcelable
import android.view.View
import android.widget.Toast
import com.zeyad.rxredux.core.view.BaseFragment
import com.zeyad.rxredux.core.viewmodel.BaseViewModel
import com.zeyad.rxredux.snackbar.SnackBarFactory

abstract class BaseFragment<S : Parcelable, VM : BaseViewModel<S>> : BaseFragment<S, VM>() {

    @JvmOverloads
    fun showToastMessage(message: String, duration: Int = Toast.LENGTH_LONG) =
            Toast.makeText(context, message, duration).show()

    /**
     * Shows a [android.support.design.widget.Snackbar] message.
     *
     * @param message An string representing a message to be shown.
     */
    fun showSnackBarMessage(view: View, message: String, duration: Int) =
            SnackBarFactory.getSnackBar(SnackBarFactory.TYPE_INFO, view, message, duration).show()

    fun showSnackBarWithAction(typeSnackBar: String,
                               view: View,
                               message: String,
                               actionText: String,
                               onClickListener: View.OnClickListener) =
            SnackBarFactory
                    .getSnackBarWithAction(typeSnackBar, view, message, actionText, onClickListener)
                    .show()

    fun showSnackBarWithAction(typeSnackBar: String,
                               view: View,
                               message: String,
                               actionText: Int,
                               onClickListener: View.OnClickListener) =
            showSnackBarWithAction(typeSnackBar, view, message, getString(actionText), onClickListener)

    /**
     * Shows a [android.support.design.widget.Snackbar] errorResult message.
     *
     * @param message  An string representing a message to be shown.
     * @param duration Visibility duration.
     */
    fun showErrorSnackBar(message: String, view: View, duration: Int) =
            SnackBarFactory.getSnackBar(SnackBarFactory.TYPE_ERROR, view, message, duration)
                    .show()
}
